event/emitter_test.go

```diff
@@ -71,9 +73,19 @@ func TestOrdering(t *testing.T) {
 	for _, msg := range msgs {
 		str := msg[0].(string)
 		if strings.HasPrefix(str, "bar") {
-			assert.Equal(t, str, <-out1)
+			assert.Equal(t, str, getTimeout(t, out1))
 		} else {
-			assert.Equal(t, str, <-out2)
+			assert.Equal(t, str, getTimeout(t, out2))
 		}
 	}
 }
+
+func getTimeout(t *testing.T, out <-chan interface{}) interface{} {
+	select {
+	case <-time.After(timeout):
```

Comment:
Will this work if the following condition is blocked on channel read?

Reply:
as in `msg := <- out` ?

Yes, that's the magic of a channel select - it will take whichever is available first